src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -265,6 +263,49 @@ class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: Acce
     }
   }
 
+  def getArbitraryPetServiceAccountKey(user: WorkbenchUser): Future[String] = {
+    for {
+      pets <- directoryDAO.getAllPetServiceAccountsForUser(user.id)
+      key <- if(pets.nonEmpty) getPetServiceAccountKey(user, pets.head.id.project) else getDefaultServiceAccountForShellProject(user)
+    } yield key
+  }
+
+  def getArbitraryPetServiceAccountToken(user: WorkbenchUser, scopes: Set[String]): Future[String] = {
+    getArbitraryPetServiceAccountKey(user).flatMap { key =>
+      getAccessTokenUsingJson(key, scopes)
+    }
+  }
+
+  private def getDefaultServiceAccountForShellProject(user: WorkbenchUser): Future[String] = {
+    val projectName = s"shell-${user.id.value}"
+    for {
+      creationOperationId <- googleBillingDAO.createProject(projectName).map(opId => Option(opId)) recover {
+        case gjre: GoogleJsonResponseException if gjre.getDetails.getCode == StatusCodes.Conflict.intValue => None
```

Comment:
this call does not always fail if the project already exists. Sometimes it still creates a project and the operation fails.

Reply:
that's insane